pkg/connector/client/helpers.go

```diff
@@ -35,17 +37,63 @@ func logBody(ctx context.Context, response *http.Response) {
 	l.Info("response body", zap.String("body", string(body)))
 }
 
-// Slack API may return errors in the response body even when the HTTP status code is 200.
-//
-//	extracts a Slack error from a Go error and wraps it with the appropriate gRPC code.
-//	This is useful when working with the slack-go library which returns plain Go errors.
+// Inspects the error returned by the slack-go library and maps it to appropriate gRPC codes.
 func WrapError(err error, contextMsg string) error {
 	if err == nil {
 		return nil
 	}
 
-	grpcCode := MapSlackErrorToGRPCCode(err.Error())
-	return uhttp.WrapErrors(grpcCode, contextMsg, err)
+	// for rate limit errors
+	var slackLibrateLimitErr *slack.RateLimitedError
+	if errors.As(err, &slackLibrateLimitErr) {
+		return uhttp.WrapErrors(codes.Unavailable, contextMsg, err)
+	}
+
+	// for 5xx status codes
+	var slackLibErr *slack.StatusCodeError
+	if errors.As(err, &slackLibErr) {
+		grpcCode := httpStatusToGRPCCode(slackLibErr.Code)
+		contextMsg = fmt.Sprintf("Slack-go API HTTP error: %s : %s", slackLibErr.Status, contextMsg)
+		return uhttp.WrapErrors(grpcCode, contextMsg, err)
+	}
+
+	// when ok: false even with 200 HTTP status codes
+	var slackErrResp *slack.SlackErrorResponse
+	if errors.As(err, &slackErrResp) {
+		grpcCode := MapSlackErrorToGRPCCode(slackErrResp.Err)
+		if len(slackErrResp.ResponseMetadata.Messages) > 0 {
+			contextMsg = fmt.Sprintf("%s (details: %v)", contextMsg, slackErrResp.ResponseMetadata.Messages)
+		}
+		if len(slackErrResp.ResponseMetadata.Warnings) > 0 {
+			contextMsg = fmt.Sprintf("%s (warnings: %v)", contextMsg, slackErrResp.ResponseMetadata.Warnings)
+		}
+		return uhttp.WrapErrors(grpcCode, contextMsg, err)
+	}
+
+	return uhttp.WrapErrors(codes.Unknown, contextMsg, err)
+}
+
+func httpStatusToGRPCCode(httpStatus int) codes.Code {
+	switch httpStatus {
+	case http.StatusBadRequest:
+		return codes.InvalidArgument
```

Comment:
We treat InvalidArgument as a user error and not as a connector error, as in for example the connector was configured incorrectly. It will not alert us to any problems with the connector, that might be correct for some APIs for http.StatusBadRequest I'm not sure if it's correct in call cases